plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/AppEngineStandardWizardPage.java

```diff
@@ -0,0 +1,172 @@
+    workspaceProjectDirectoryButtonPosition.horizontalSpan = 3;
+    workspaceProjectDirectoryButton.setLayoutData(workspaceProjectDirectoryButtonPosition);
+
+    customProjectDirectoryButton = new Button(projectDirectoryGroup, SWT.RADIO);
+    customProjectDirectoryButton.setSelection(false); // not by default
+    customProjectDirectoryButton.setText("Create new project in:");
+    customProjectDirectoryButton.addSelectionListener(directorySwitcher);
+    GridData customProjectDirectoryButtonPosition = new GridData();
+    customProjectDirectoryButtonPosition.horizontalAlignment = GridData.FILL;
+    customProjectDirectoryButtonPosition.grabExcessHorizontalSpace = true;
+    customProjectDirectoryButtonPosition.horizontalSpan = 3;
+    customProjectDirectoryButton.setLayoutData(customProjectDirectoryButtonPosition);
+
+    Label projectDirectoryLabel = new Label(projectDirectoryGroup, SWT.NONE);
+    projectDirectoryLabel.setText("Directory:");
+
+    projectDirectoryField = new Text(projectDirectoryGroup, SWT.BORDER);
+    projectDirectoryField.setEnabled(false);
+    GridData projectDirectoryFieldPosition = new GridData();
+    projectDirectoryFieldPosition.horizontalAlignment = GridData.FILL;
+    projectDirectoryFieldPosition.grabExcessHorizontalSpace = true;
+    projectDirectoryField.setLayoutData(projectDirectoryFieldPosition);
+
+    projectDirectoryBrowseButton = new Button(projectDirectoryGroup, SWT.NONE);
+    projectDirectoryBrowseButton.setEnabled(false);
+    projectDirectoryBrowseButton.setText("Browse...");
+    projectDirectoryBrowseButton.addSelectionListener(new ProjectDirectoryPicker());
+  }
+  
+  private final class ProjectDirectoryPicker extends SelectionAdapter {
+    @Override
+    public void widgetSelected(SelectionEvent event) {
+      DirectoryDialog dialog = new DirectoryDialog(getShell(), SWT.OPEN);
+      dialog.setMessage("Choose a directory for the project contents:");
+      String userChoice = dialog.open();
+      if (userChoice != null) {
+        projectDirectoryField.setText(userChoice);
+      }
+    }
+  }
+
+  private final class PageValidator implements ModifyListener {
+    @Override
+    public void modifyText(ModifyEvent event) {
+      // todo add checks for project ID and directory 
+      // todo add error messages
+      boolean complete = JavaPackageValidator.validate(javaPackageField.getText()) 
+          && EclipseProjectNameValidator.validate(eclipseProjectNameField.getText());
+      setPageComplete(complete);
+    }
+  }  
+  
+  private final class DirectorySwitcher extends SelectionAdapter {
+    @Override
+    public void widgetSelected(SelectionEvent event) {
+      projectDirectoryBrowseButton.setEnabled(event.widget == customProjectDirectoryButton);
+      projectDirectoryField.setEnabled(event.widget == customProjectDirectoryButton);
+    }
+  }
+
+  AppEngineStandardProjectConfig getAppEngineStandardProjectConfig() {
```

Comment:
The best practice with wizards is to have the wizard instance host the data, and have the pages pull out and set that data.  It avoids having wizard pages needing to access each other's contents.  For example, when we add support for managed APIs, the ManagedApiWizardPage could need to know the Project Id to enable the API, and the Eclipse project to fetch and install the API jars.